src/layout/FileUpload/shared/summary.ts

```diff
@@ -0,0 +1,52 @@
+import { useAppSelector } from 'src/common/hooks/useAppSelector';
+import type { IFormData } from 'src/features/form/data';
+import type { IAttachment, IAttachments } from 'src/shared/resources/attachments';
+import type { LayoutNodeFromType } from 'src/utils/layout/hierarchy.types';
+
+export function extractListFromBinding(formData: IFormData, listBinding: string): string[] {
+  return Object.keys(formData)
+    .filter((key) => key.startsWith(listBinding))
+    .map((key) => formData[key]);
+}
+
+export function attachmentsFromUuids(componentId: string, uuids: string[], attachments: IAttachments): IAttachment[] {
+  if (!uuids.length) {
+    return [];
+  }
+
+  return uuids
+    .map((uuid) => {
+      const attachmentsForComponent = attachments[componentId];
+      if (attachmentsForComponent) {
+        const foundAttachment = attachmentsForComponent.find((a) => a.id === uuid);
+        if (foundAttachment && foundAttachment.name) {
+          return foundAttachment;
+        }
+      }
+
+      return null;
+    })
+    .filter((a) => a !== null) as IAttachment[];
```

Comment:
What do you think about refactoring a little to avoid filtering null and casting the type? Would it be a little more readable to follow the suggestion? The suggestion has moved `attachmentsForComponent `outside the loop to avoid creating and assigning on every iteration. :) 

```suggestion
  const attachmentsForComponent = attachments[componentId];

  if (!attachmentsForComponent) {
    return [];
  }

  let componentAttachments: IAttachment[] = [];

  uuids.forEach((uuid) => {
    const foundAttachment = attachmentsForComponent.find((a) => a.id === uuid);
    if (foundAttachment?.name) {
      componentAttachments = [...componentAttachments, foundAttachment];
    }
  });

  return componentAttachments;
```

Reply:
Finally looked at closely at this, and yup, that's much better! I took the liberty to rewrite it to use `for (const uuid of uuids)` and `componentAttachments.push()` as well. 🥳 Thank you!

Reply:
No worries! Sounds great to use the push method, then componentAttachments can be const instead of let. Great work! 🥳